### PR TITLE
stress test improvement

### PR DESCRIPTION
**Pull request by meldafrawi**
Signed-off-by: Mohamed Eldafrawi <mohamed.eldafrawi@rancher.com>

### CODE REVIEW

File: manager/integration/tests/test_stress.py [hunk 1]
```diff
@@ -145,6 +147,43 @@ def snapshot_create_and_record_md5sum(client, core_api, volume_name, pod_name, s
...
+
+    delete_and_wait_pod(core_api, pod_name)
+
+    wait_for_volume_detached(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    volume.attach(hostId=host_id, disableFrontend=True)
+
+    wait_for_volume_healthy_no_frontend(client, volume_name)
```
Comment (meldafrawi) [inline manager/integration/tests/test_stress.py]: wait_for_volume_healthy_no_frontend return volume object.

File: manager/integration/tests/test_stress.py [hunk 2]
```diff
@@ -313,6 +315,24 @@ def restore_and_check_random_backup(client, core_api, volume_name, pod_name, sna
...
+
+    replica_name = volume["replicas"][replica_id]["name"]
+
+    volume.replicaRemove(name=replica_name)
+
+    wait_for_volume_replica_count(client, volume_name, replica_count)
+
+    replica_names = map(lambda replica: replica.name, volume["replicas"])
+
+    wait_new_replica_ready(client, volume_name, replica_names)
```
Comment (meldafrawi) [inline manager/integration/tests/test_stress.py]: - randomly decide if we want to wait to wait for replica to be rebuild.
- add a parameter to configure the chance to wait for an action to be completed.

File: manager/integration/tests/test_stress.py [hunk 3]
```diff
@@ -492,26 +492,56 @@ def generate_load(request):
...
-    create_snapshot(longhorn_api_client, volume_name)
-
-    delete_random_snapshot(longhorn_api_client, volume_name)
+    create_recurring_jobs(longhorn_api_client, volume_name)
 
-    # execute 3 more random actions
-    for round in range(3):
-        action = randrange(0, 4)
+    # execute 10 random actions
+    for round in range(10):
```
Comment (meldafrawi) [inline manager/integration/tests/test_stress.py]: add parameter for action count.

**Comment by yasker** [comment]
Merged. Let's start running it.

**Pull request merged**